### PR TITLE
Add dependencies to cyclonedx

### PR DESCRIPTION
**Pull request by spiffcs**
Updated CycloneDX relationships based off of #706

### CODE REVIEW

**Comment by hectorj2f** [comment]
Lgtm

File: internal/formats/common/cyclonedxhelpers/format.go [hunk 1]
```diff
@@ -97,18 +105,63 @@ func toBomDescriptor(name, version string, srcMetadata source.Metadata) *cyclone
 	}
 }
 
+func lookupRelationship(ty artifact.RelationshipType) bool {
```
Comment (wagoodman) [inline internal/formats/common/cyclonedxhelpers/format.go:108]: nit: the name of this function doesn't really convey what its purpose is, which is to filter down relationship types to those that are meant for package dependencies

File: internal/formats/common/cyclonedxhelpers/format.go [hunk 2]
```diff
@@ -97,18 +105,63 @@ func toBomDescriptor(name, version string, srcMetadata source.Metadata) *cyclone
...
+	switch ty {
+	case artifact.OwnershipByFileOverlapRelationship:
```
Comment (wagoodman) [inline internal/formats/common/cyclonedxhelpers/format.go:110]: I'm not 100% sure that this relationship belongs here, but I could be wrong

Comment (spiffcs) [reply internal/formats/common/cyclonedxhelpers/format.go:110]: I can remove this one for now.

**Comment by spiffcs** [comment]
@wagoodman updated based on the PR feedback

File: internal/formats/common/cyclonedxhelpers/format.go (hunk 1 above)
Comment (wagoodman) [reply internal/formats/common/cyclonedxhelpers/format.go]: That's better; what about something like `isExpressiblePackageRelationship`? Beyond naming, I think a helpful comment to help future devs figure which qualities of a relationship indicate that they should be included here.

File: internal/formats/common/cyclonedxhelpers/format.go [hunk 3]
```diff
@@ -97,18 +105,61 @@ func toBomDescriptor(name, version string, srcMetadata source.Metadata) *cyclone
...
+	}
+	return false
+}
+
+func toDependencies(relationships []artifact.Relationship) []cyclonedx.Dependency {
+	result := make([]cyclonedx.Dependency, 0)
+	for _, r := range relationships {
+		exists := relationshipTypeExists(r.Type)
+		if !exists {
+			log.Warnf("unable to convert relationship from CycloneDX 1.3 JSON, dropping: %+v", r)
```
Comment (wagoodman) [inline internal/formats/common/cyclonedxhelpers/format.go:127]: nit: this warning indicates a specific CycloneDX version, but this helper could be used for any CycloneDX version

**Comment by spiffcs** [comment]
looks like my changes to the tests did not get factored in

**Comment by wagoodman** [comment]
> looks like my changes to the tests did not get factored in

also other changes didn't survive the rebase:
```
...
func lookupRelationship(ty artifact.RelationshipType) bool {
...
```

**Review by wagoodman — APPROVED** [review]
survived the rebase!

**Comment by hectorj2f** [comment]
Thanks guys 👏🏻

**Pull request merged**